lib/puppetserver/ca/config/puppet.rb

```diff
@@ -46,20 +46,33 @@ def initialize(supplied_config_path = nil)
         # on Windows are unsupported.
         # Note that Puppet Server runs as the [pe-]puppet user but to
         # start/stop it you must be root.
-        def user_specific_conf_dir
-          @user_specific_conf_dir ||=
-            if Puppetserver::Ca::Utils::Config.running_as_root?
+        def running_as_root?
+          @as_root ||= Puppetserver::Ca::Utils::Config.running_as_root?
+        end
+
+        def user_specific_puppet_confdir
+          @user_specific_puppet_confdir ||=
+            if running_as_root?
               '/etc/puppetlabs/puppet'
             else
               "#{ENV['HOME']}/.puppetlabs/etc/puppet"
             end
         end
 
-        def user_specific_conf_file
-          user_specific_conf_dir + '/puppet.conf'
+        def user_specific_puppetconf_file
+          user_specific_puppet_confdir + '/puppet.conf'
+        end
+
+        def user_specific_puppetserver_confdir
+          @user_specific_puppetserver_confdir ||=
+            if running_as_root?
+              '/etc/puppetlabs/puppetserver'
+            else
+              "#{ENV['HOME']}/.puppetserver"
```

Comment:
It's already a thing, I had it locally. I've been looking into it more and I think it we can safely set it to `~/.puppetlabs` (especially since we're doing this in 7) but we should probably update some _really_ old docs in Puppet Server [here](https://github.com/puppetlabs/puppetserver/blob/6.x/documentation/gems.markdown#installing-and-removing-gems).